test/f90_correct/src/string_array_pointer.f90

```diff
@@ -0,0 +1,19 @@
+!
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+
+! Check that the type length of a string array pointer is set correctly.
+
+program string_array_pointer_test
+  implicit none
+  character(:), pointer :: ptr(:)
+  character(3), target :: array(1) = ['foo']
+  ptr(2:2) => array
+  ptr(2) = 'bar'
+  if (len(ptr) /= 3) stop 1
```

Comment:
I feel slightly confused here. Is this the len value we care about in the #1008? IMHO we should check for length of the pointed string, namely:
```
  if (len(ptr(2)) /= 3) stop 1
```

Reply:
The code is correct and equivalent to what you suggest.

The Fortran 2018 standard states (16.9.112, "LEN (STRING [, KIND])"):
> **Result Value.** The result has a value equal to the number of characters in STRING if it is scalar or in an element of STRING if it is an array.